internal/monitors/statsd/parser_test.go

```diff
@@ -0,0 +1,84 @@
+package statsd
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseFields(t *testing.T) {
+	cases := []struct {
+		pattern        string
+		substrs        []string
+		startWithField bool
+		expectNil      bool
+	}{
+		{
+			"metric.count",
+			[]string{"metric.count"},
+			false,
+			false,
+		},
+		{
+			"metric.count{",
+			nil,
+			false,
+			true,
+		},
+		{
+			"{metric.count",
+			nil,
+			false,
+			true,
+		},
+		{
+			"metric.count}",
+			nil,
+			false,
+			true,
+		},
+		{
+			"{{metric.count}}",
+			nil,
+			false,
+			true,
+		},
+		{
+			"{metric.count}",
+			[]string{"metric.count"},
+			true,
+			false,
+		},
+		{
+			"cluster.cds_{traffic}_{mesh}_{service}-vn_{}.{action}",
+			[]string{"cluster.cds_", "traffic", "_", "mesh", "_", "service", "-vn_", "", ".", "action"},
+			false,
+			false,
+		},
+		{
+			"{cluster}.cds_{traffic}_{mesh}_{service}-vn_{}.{action}",
+			[]string{"cluster", ".cds_", "traffic", "_", "mesh", "_", "service", "-vn_", "", ".", "action"},
+			true,
+			false,
+		},
+		{
+			// Cannot have back-to-back patterns
+			"{cluster}.cds_{traffic}{mesh}_{service}-vn_{}.{action}",
+			nil,
+			false,
+			true,
+		},
```

Comment:
Can we have a test case with for a string not ending with a {pattern}? like `metric.{region}.count`?

Reply:
Added two cases with only inner dynamic fields, one with a pattern that starts with a dynamic field, and another that doesn't.

Reply:
Cool, thanks!